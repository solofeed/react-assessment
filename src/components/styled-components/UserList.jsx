import styled from 'styled-components';

export const Row = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;
  border: 1px solid #000;
  padding: 20px;
  
  span {
    display: block;
    margin-bottom: 5px;
  }
`;

export const UserInfo = styled.div`
  border-right: 1px solid #000;
  text-align: left;
  width: 260px;
`;

export const Users = styled.div`
  max-height: 300px;
  overflow: scroll;
  margin-top: 15px;
`;
