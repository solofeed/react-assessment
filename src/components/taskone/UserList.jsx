import React, { useState } from 'react';
import { useDebounce } from '../../hooks/useDebounce';
import { useFetchUsers } from '../../hooks/useFetchUsers';
import { UserRow } from './UserRow';
import { Users } from '../styled-components/UserList';

const UserList = () => {
  const [filter, setFilter] = useState('');
  const debouncedFilter = useDebounce(filter);
  const users = useFetchUsers(debouncedFilter);

  return (
    <div>
      <div>
        Filter:
        <input
          type="text"
          onChange={(e) => setFilter(e.target.value)}
          value={filter}
          placeholder="Enter username"
        />
      </div>
      <Users>
        {users.map((user) => (
          <UserRow
            key={user.id}
            id={user.id}
            name={user.name}
            username={user.username}
            address={user.address}
            email={user.email}
            phone={user.phone}
          />
        ))}
      </Users>
    </div>
  );
};

export default UserList;
