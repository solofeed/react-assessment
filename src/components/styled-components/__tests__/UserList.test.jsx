import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { Row, UserInfo, Users } from '../UserList';

describe('UserList snapshots', () => {
  it('Row renders correctly', () => {
    const { container } = render(<Row />);
    expect(container).toMatchSnapshot();
  });

  it('UserInfo renders correctly', () => {
    const { container } = render(<UserInfo />);
    expect(container).toMatchSnapshot();
  });

  it('Users renders correctly', () => {
    const { container } = render(<Users />);
    expect(container).toMatchSnapshot();
  });
});
