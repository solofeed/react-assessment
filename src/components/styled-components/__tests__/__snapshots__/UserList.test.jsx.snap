// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList snapshots Row renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 10px;
  border: 1px solid #000;
  padding: 20px;
}

.c0 span {
  display: block;
  margin-bottom: 5px;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`UserList snapshots UserInfo renders correctly 1`] = `
.c0 {
  border-right: 1px solid #000;
  text-align: left;
  width: 260px;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`UserList snapshots Users renders correctly 1`] = `
.c0 {
  max-height: 300px;
  overflow: scroll;
  margin-top: 15px;
}

<div>
  <div
    class="c0"
  />
</div>
`;
